res/qml/Library.qml

```diff
@@ -8,22 +9,76 @@ Item {
         color: Theme.deckBackgroundColor
         anchors.fill: parent
 
+        LibraryControl {
+            id: libraryControl
+
+            onMoveSelection: listView.moveSelection(offset)
+            onLoadSelectedTrack: listView.loadSelectedTrack(group, play)
+            onLoadSelectedTrackIntoNextAvailableDeck: listView.loadSelectedTrackIntoNextAvailableDeck(play)
+        }
+
         ListView {
+            id: listView
+
+            function moveSelection(value) {
+                if (value == 0)
+                    return ;
+
+                const rowCount = model.rowCount();
+                if (rowCount == 0)
+                    return ;
+
+                let newIndex = currentIndex = (currentIndex + value) % rowCount;
+                while (newIndex < 0)newIndex += rowCount
+                currentIndex = newIndex;
```

Comment:
took me quite some time to realize that this is just mathematically correct definition of the modulus. Could you instead factor this out into a utility function?

Reply:
Modulo can be negative, this ensures it's always positive. I can move it to mathutils.

Reply:
Well, that is the case in most programming languages and hardware implementations. I just learned that the mathematical definition includes that or the modulus operation `n = q % s`, 0 < n < s and s > 0. So we should probably encapsulate the mathematically correct definition in a function in mathutils.